src/data/equiv/mul_add.lean

```diff
@@ -314,6 +314,29 @@ def monoid_hom_congr {M N P Q} [mul_one_class M] [mul_one_class N] [comm_monoid
   right_inv := λ k, by { ext, simp, },
   map_mul' := λ h k, by { ext, simp, }, }
 
+/-- A family of multiplicative equivalences `Π j, (Ms j ≃* Ns j)` generates a
+multiplicative equivalence between `Π j, Ms j` and `Π j, Ns j`.
+
+This is the `mul_equiv` version of `equiv.Pi_congr_right`.
+-/
+@[to_additive add_equiv.Pi_congr_right "A family of additive equivalences `Π j, (Ms j ≃+ Ns j)` generates an
+additive equivalence between `Π j, Ms j` and `Π j, Ns j`.
+
+This is the `add_equiv` version of `equiv.Pi_congr_right.`"]
+def Pi_congr_right {η : Type*}
+  {Ms Ns : η → Type*} [Π j, monoid (Ms j)] [Π j, monoid (Ns j)]
+  (es : ∀ j, Ms j ≃* Ns j) : (Π j, Ms j) ≃* (Π j, Ns j) :=
+{ map_mul' := λ x y,
+  by { ext j,
+       simp_rw [equiv.Pi_congr_right, pi.mul_apply, mul_equiv.coe_to_equiv, mul_equiv.map_mul] },
+  .. equiv.Pi_congr_right (λ j, (es j).to_equiv) }
+
+@[simp, to_additive add_equiv.Pi_congr_right_apply]
+lemma Pi_congr_right_apply {η : Type*}
+  {Ms Ns : η → Type*} [Π j, monoid (Ms j)] [Π j, monoid (Ns j)]
+  (es : ∀ j, Ms j ≃* Ns j) (x j) :
+  mul_equiv.Pi_congr_right es x j = es j (x j) := rfl
```

Comment:
Golfs to:
```suggestion
def Pi_congr_right {η : Type*}
  {Ms Ns : η → Type*} [Π j, mul_one_class (Ms j)] [Π j, mul_one_class (Ns j)]
  (es : Π j, Ms j ≃* Ns j) : (Π j, Ms j) ≃* (Π j, Ns j) :=
{ to_fun := λ x j, es j (x j),
  map_mul' := λ x y, funext $ λ j, (es j).map_mul  _ _,
  .. equiv.Pi_congr_right (λ j, (es j).to_equiv) }
```
By supplying `to_fun` the proof is easier to work out and `simp`s does the right thing.